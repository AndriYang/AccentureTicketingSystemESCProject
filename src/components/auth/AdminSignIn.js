import React, { Component } from 'react'
import { connect } from 'react-redux'
import { adminsignIn } from '../../store/actions/authActions'
import { Redirect } from 'react-router-dom'

export class AdminSignIn extends Component {
  state = {
    email: '',
    emailError: false,
    password: '',
    passwordError: false,
    count: 1,
    countError: false,
    // time:0,
    // timeError: false
  }

  handleChange = (e) => {
    this.setState({
      [e.target.id]: e.target.value
    })
    if(e.target.id==='email'){
    this.validateEmail(e.target.value);
   }

   if(e.target.id==='password'){
    if(e.target.value.length < 6 || e.target.value===null){
      this.setState({
        passwordError:true
      })
    } else {
      this.setState({
        passwordError:false,
        password:e.target.value
      })
    }
 }
  }

  handleSubmit = (e) => {
    e.preventDefault();
    console.log(this.state.count);
    this.props.adminsignIn(this.state);
    if (this.state.count>2){
      this.setState({
        countError: true
      })
    }
    if (this.state.count <=3) {
      this.setState({
        count: this.state.count+1
      })
    }
    // const createdAt= new Date();
    // console.log(createdAt.getMinutes());
    // if (this.state.time==0){
    //   this.setState({
    //     time: createdAt.getMinutes()
    //   })
    // }
    // if (this.state.time !=0) {
    //   const elapse = (createdAt.getMinutes()-this.state.time)<2;
    //
    //   this.setState({
    //     timeError: elapse
    //   })
    // }
    // console.log("timeError: " + this.state.timeError);
    // console.log("time: " + this.state.time);

  }

  validateEmail(email){
   const pattern = /[a-zA-Z0-9]+[\.]?([a-zA-Z0-9]+)?[\@][a-z]{3,9}[\.][a-z]{2,5}/g;
   const result = pattern.test(email);
   if(result===true){
     this.setState({
       emailError:false,
       email : email
     })
   }else{
     this.setState({
       emailError:true
     })
   }
 }

  render() {
    const { authError, auth } = this.props;
    const {email, password} = this.state;
    const isEnabled = this.state.emailError == false && password.length >5 && this.state.countError == false;
    if (auth.uid) return<Redirect to='/dashboard' />

    return (
      <div className="container">
        <form onSubmit={this.handleSubmit} className="white formQuery" id="adminLoginForm">
          <h5 className="text-white  purple darken-3">Sign in</h5>
          <div className="input-field">
            <label className="grey-text text-darken-3" htmlFor="email" name="emaillable">someone@accenture.com</label>
            <input type="email" name= "email" id= "email"  class="validate" onChange={this.handleChange}/>
            {this.state.emailError ? <span style={{color: "red"}}>Please Enter valid email address</span> : ''}
          </div>
          <div className="input-field">
            <label className="grey-text text-darken-3" htmlFor="password">Password</label>
            <input type="password" id="password" class="validate" onChange={this.handleChange}/>
            {this.state.passwordError ? <span style={{color: "red"}}>Wrong Password</span> : ''}
          </div>
          <div className="input-field">
            <button disabled={!isEnabled} className="btn purple darken-3 text-white z-depth-0" id="button">Login</button>
            <div className="red-text center">
              {this.state.countError ? <span style={{color: "red"}}>Denied Login. Please contact the security!</span> : ''}
              { authError ? <p> {authError} </p> : null }
            </div>
          </div>
        </form>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  //console.log(state);
  return {
    authError: state.auth.authError,
    auth: state.firebase.auth
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    adminsignIn: (creds) => dispatch(adminsignIn(creds))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(AdminSignIn)
