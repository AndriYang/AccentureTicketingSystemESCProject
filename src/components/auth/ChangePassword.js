import React, { Component } from 'react'
import { connect } from 'react-redux'
import { changePassword } from '../../store/actions/authActions'
import { Redirect } from 'react-router-dom'

export class ChangePassword extends Component {
  state = {
    newPassword: '',
    password: ''
  }

  handleChange = (e) => {
    this.setState({
      [e.target.id]: e.target.value
    })
  }

  handleSubmit = (e) => {
    e.preventDefault();
    this.props.changePassword(this.state);
    this.props.history.push('/dashboard')
  }



  render() {

    const { authError, auth } = this.props;
    if (!auth.uid) return<Redirect to='/' />

    return(
      <div className="container">
        <form onSubmit={this.handleSubmit} className="white" id='loginForm'>
        <div className="input-field">
          <label htmlFor="password"> Current Password</label>
          <input type="password" id="password" onChange={this.handleChange}/>
        </div>
        <div className="input-field">
          <label htmlFor="password"> New Password</label>
          <input type="password" id="newPassword" onChange={this.handleChange}/>
        </div>

        <div className="input-field">
          <button className="btn purple darken-3 text-white">Confirm</button>
        </div>
        </form>
      </div>

    )

  }
}



const mapStateToProps = (state) => {
  return {
    authError: state.auth.authError,
    auth: state.firebase.auth
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    changePassword: (creds) => dispatch(changePassword(creds))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ChangePassword)
