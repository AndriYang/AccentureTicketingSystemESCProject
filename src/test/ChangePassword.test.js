import React from 'react';
import {shallow} from 'enzyme';
import {ChangePassword} from '../components/auth/ChangePassword';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store'

configure({ adapter: new Adapter() });

  test('When the form is submitted', ()=>{
    const mockLoginfn = {push: jest.fn()};
    const change =jest.fn();
    const comp =(
      <ChangePassword history={mockLoginfn} changePassword={change}/>
    );
    const wrapper = shallow(comp);
    const oldPassword = wrapper.find('input').at(0);
    const newPassword = wrapper.find('input').at(1);
    const submitForm = wrapper.find('form');

    oldPassword.simulate('change', {target:
      {id: 'password', value: 'test1234'}
    })

    newPassword.simulate('change', {target:
      {id: 'newPassword', value: '1234test'}
    })

    submitForm.simulate('submit', {preventDefault() {}})
    expect(change.mock.calls[0][0]).toEqual(
      {password: 'test1234', newPassword: '1234test'}
    )
  })

  it('Check password setState ', ()=> {
    const mockLoginfn = {push: jest.fn()};
    const change =jest.fn();
    const comp =(
      <ChangePassword history={mockLoginfn} changePassword={change}/>
    );
    const wrapper = shallow(comp);
    const oldPassword = wrapper.find('input').at(0);
    oldPassword.simulate('change', {target:
      {id: 'password', value: 'test1234'}
    })

    expect(wrapper.state('password')).toEqual('test1234');
  })

  it('Check new password setState ', ()=> {
    const mockLoginfn = {push: jest.fn()};
    const change =jest.fn();
    const comp =(
      <ChangePassword history={mockLoginfn} changePassword={change}/>
    );
    const wrapper = shallow(comp);
    const oldPassword = wrapper.find('input').at(0);
    oldPassword.simulate('change', {target:
      {id: 'newPassword', value: '1234test'}
    })

    expect(wrapper.state('newPassword')).toEqual('1234test');
  })

  it('print the initial state',() => {
    const data = new ChangePassword();
    expect(data).toMatchSnapshot();
  })

  it('print the state',() => {
    const data = new ChangePassword({password: 'test1234',
    newPassword: '1234test'});

    expect(data).toMatchSnapshot();
  })
