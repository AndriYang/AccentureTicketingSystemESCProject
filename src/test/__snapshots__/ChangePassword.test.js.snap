// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`print the initial state 1`] = `
ChangePassword {
  "context": undefined,
  "handleChange": [Function],
  "handleSubmit": [Function],
  "props": undefined,
  "refs": Object {},
  "state": Object {
    "newPassword": "",
    "password": "",
  },
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;

exports[`print the state 1`] = `
ChangePassword {
  "context": undefined,
  "handleChange": [Function],
  "handleSubmit": [Function],
  "props": Object {
    "newPassword": "1234test",
    "password": "test1234",
  },
  "refs": Object {},
  "state": Object {
    "newPassword": "",
    "password": "",
  },
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;
